memgpt/config.py

```diff
@@ -104,70 +131,38 @@ def load(cls) -> "MemGPTConfig":
-            archival_storage_type, archival_storage_path, archival_storage_uri = "local", None, None
-            if "archival_storage" in config:
-                archival_storage_type = config.get("archival_storage", "type")
-                archival_storage_path = config.get("archival_storage", "path") if config.has_option("archival_storage", "path") else None
-                archival_storage_uri = config.get("archival_storage", "uri") if config.has_option("archival_storage", "uri") else None
-
-            anon_clientid = config.get("client", "anon_clientid")
-
-            return cls(
-                model=model,
-                context_window=context_window,
-                preset=preset,
-                model_endpoint=model_endpoint,
-                default_persona=default_persona,
-                default_human=default_human,
-                default_agent=default_agent,
-                openai_key=openai_key,
-                azure_key=azure_key,
-                azure_endpoint=azure_endpoint,
-                azure_version=azure_version,
-                azure_deployment=azure_deployment,
-                azure_embedding_deployment=azure_embedding_deployment,
-                embedding_model=embedding_model,
-                embedding_dim=embedding_dim,
-                embedding_chunk_size=embedding_chunk_size,
-                archival_storage_type=archival_storage_type,
-                archival_storage_path=archival_storage_path,
-                archival_storage_uri=archival_storage_uri,
-                anon_clientid=anon_clientid,
-                config_path=config_path,
-            )
-
+            config_dict = {
+                "model": get_field(config, "model", "model"),
+                "model_endpoint": get_field(config, "model", "model_endpoint"),
+                "model_endpoint_type": get_field(config, "model", "model_endpoint_type"),
+                "model_wrapper": get_field(config, "model", "model_wrapper"),
+                "context_window": get_field(config, "model", "context_window"),
+                "preset": get_field(config, "defaults", "preset"),
+                "persona": get_field(config, "defaults", "persona"),
+                "human": get_field(config, "defaults", "human"),
+                "agent": get_field(config, "defaults", "agent"),
+                "openai_key": get_field(config, "openai", "key"),
+                "azure_key": get_field(config, "azure", "key"),
+                "azure_endpoint": get_field(config, "azure", "endpoint"),
+                "azure_version": get_field(config, "azure", "version"),
+                "azure_deployment": get_field(config, "azure", "deployment"),
+                "azure_embedding_deployment": get_field(config, "azure", "embedding_deployment"),
+                "embedding_endpoint": get_field(config, "embedding", "embedding_endpoint"),
+                "embedding_endpoint_type": get_field(config, "embedding", "embedding_endpoint_type"),
```

Comment:
Should this be:
```python
"embedding_endpoint": get_field(config, "embedding", "endpoint"),
"embedding_endpoint_type": get_field(config, "embedding", "endpoint_type"),
```

I think this is causing the bug in https://learningsystems.slack.com/archives/D0199P12REE/p1699925736563559

Reply:
Also it seems like there's some inconsistency in naming here:

```
[model]
model_endpoint = http://127.0.0.1:1234
model_endpoint_type = lmstudio
model_wrapper = airoboros-l2-70b-2.1
context_window = 8192

[embedding]
endpoint_type = local
dim = 384
chunk_size = 300
```

Seems like it should either be:
```
[model]
model_* = ...

[embedding]
embedding_* = ...
```

Or:
```
[model]
* = ...

[embedding]
* = ...
```

Reply:
ah yeah I this is a bug, I forgot to rename the embedding-related fields when writing the config

Reply:
This should be fixed now